src/components/users/Accounts.jsx

```diff
@@ -63,14 +93,39 @@ const CreateAccount = ({
     setAccounts,
 }) => {
     const [fullName, setFullName] = useState(accounts.fullName);
+    const [_userAccount, _setUserAccount] = useState(accounts.userAccount);
     const [userAccount, setUserAccount] = useState(accounts.userAccount);
+    const [userAccountInvalidHint, setUserAccountInvalidHint] = useState("");
+    const [isUserAccountValid, setIsUserAccountValid] = useState(null);
     const [password, setPassword] = useState(accounts.password);
     const [confirmPassword, setConfirmPassword] = useState(accounts.confirmPassword);
     const [isPasswordValid, setIsPasswordValid] = useState(false);
 
     useEffect(() => {
-        setIsUserValid(isPasswordValid && userAccount.length > 0);
-    }, [setIsUserValid, isPasswordValid, userAccount]);
+        debounce(300, () => setUserAccount(_userAccount))();
+    }, [_userAccount, setUserAccount]);
+
+    useEffect(() => {
+        setIsUserValid(isPasswordValid && isUserAccountValid);
+    }, [setIsUserValid, isPasswordValid, isUserAccountValid]);
+
+    useEffect(() => {
+        let valid = true;
+        setUserAccountInvalidHint("");
+        if (userAccount.length === 0) {
+            valid = null;
+        } else if (userAccount.length > 32) {
+            valid = false;
+            setUserAccountInvalidHint(_("The user name is too long"));
+        } else if (reservedNames.includes(userAccount)) {
+            valid = false;
+            setUserAccountInvalidHint(_("Sorry, that user name is not available. Please try another."));
```

Comment:
Since this is blocked exclusively because it is reserved word, it might be nice to say that specifically here.

```suggestion
            setUserAccountInvalidHint(_("This is a reserved word; please try another user name"));
```

Reply:
Updated